app/src/main/java/tech/jpco/qen/viewModel/VMDataClasses.kt

```diff
@@ -0,0 +1,44 @@
+package tech.jpco.qen.viewModel
+
+sealed class MetaEvent {
+    object CyclePage : MetaEvent()
+    object ClearPage : MetaEvent()
+    data class NewPage(val ar: Float) : MetaEvent()
+    data class SelectPage(val page: Int) : MetaEvent()
+    data class CurrentPage(val ar: Float) : MetaEvent()
+}
+
+//TODO refactor content and ratio into normal data class implementation?
+data class SelectedPage(val current: Int, val total: Int) {
+    var content: List<DrawPoint> = listOf()
+        private set
+
+    var ratio: Float = 0f
+        private set
+
+    constructor(current: Int, total: Int, content: List<DrawPoint>, ratio: Float) : this(current, total) {
+        this.content = content
+        this.ratio = ratio
+    }
+
+    override fun toString(): String =
```

Comment:
It may well smell, but the custom `toString` gets me the data I excluded from the default constructor (fixed by going to a more conventional data class, see below), the List's size instead of potentially having a giant list of individual points, and the identityHashCode for log-debugging purposes. I can see why it would be a smell, but IMO the functionality trumps the smell, so I'm curious if you still think it should be changed.

Reply:
I think that justification is fair, logging a very large list makes the logging less useful.